docs/quickstart.rst

```diff
@@ -61,182 +166,269 @@ instance had already existing in the destination directory, this command
-streaming WAL contents from the primary.
+For ``pg_hba.conf`` on the monitor node pg_autoctl inspects the local network
+and makes its best guess about the subnet to allow. In our case it guessed
+correctly:
+
+.. code-block:: ini
+
+   # automatically added to the monitor
+
+   host "pg_auto_failover" "autoctl_node" 10.0.1.0/24 trust
+
+If worker nodes have more ad-hoc addresses and are not in the same subnet, it's
+better to disable pg_autoctl's automatic modification of pg_hba using the
+``--skip-pg-hba`` command line option during creation. You will then need to
+edit the hba file by hand. Another reason for manual edits would be to use
+special authentication methods.
 
 Watch the replication
 ---------------------
 
 First let’s verify that the monitor knows about our nodes, and see what
 states it has assigned them:
 
-.. code-block:: text
+.. code-block::
 
-   pg_autoctl show state --pgdata ./monitor
-        Name |   Port | Group |  Node |     Current State |    Assigned State
-   ----------+--------+-------+-------+-------------------+------------------
-   127.0.0.1 |   6010 |     0 |     1 |           primary |           primary
-   127.0.0.1 |   6011 |     0 |     2 |         secondary |         secondary
+   ssh -l ha-admin `vm_ip monitor` pg_autoctl show state --pgdata monitor
+
+                              Name |   Port | Group |  Node |     Current State |    Assigned State
+   --------------------------------+--------+-------+-------+-------------------+------------------
+   ha-demo-a.internal.cloudapp.net |   5432 |     0 |     1 |           primary |           primary
+   ha-demo-b.internal.cloudapp.net |   5432 |     0 |     2 |         secondary |         secondary
 
-This looks good. We can add data to the primary, and watch it get
-reflected in the secondary.
+
+This looks good. We can add data to the primary, and later see it appear in the
+secondary. We'll connect to the database from inside our "app" virtual machine,
+using a connection string obtained from the monitor. First we'll get the
+connection string and store it in a local environment variable:
+
+.. code-block:: bash
+
+   APP_DB_URI=$( \
+     ssh -l ha-admin `vm_ip monitor` \
+       pg_autoctl show uri --formation default --pgdata monitor \
+   )
+
+The connection string contains both our nodes, comma separated, and includes
+the url parameter ``?target_session_attrs=read-write`` telling psql that we
+want to connect to whichever of these servers supports reads *and* writes.
+That will be the primary server.
 
 .. code-block:: bash
 
-   # add data to primary
-   psql -p 6010 \
-     -c 'create table foo as select generate_series(1,1000000) bar;'
+   # connect to database via psql on the app vm
+   ssh -l ha-admin -t `vm_ip app` psql "\"$APP_DB_URI\""
+
+Within psql:
 
-   # query secondary
-   psql -p 6011 -c 'select count(*) from foo;'
-     count
-   ---------
-    1000000
+.. code-block:: psql
```

Comment:
I would have preferred to send this SQL statement in one step using `psql -c`, but bash was grabbing the `-c`. Perhaps there's a way to quote the ssh command better to fix that.

Reply:
I guess you could alway `echo 'select ...' | ssh ... sql` if nothing else works?

Reply:
You can probably use `ssh <connection options> -- psql -c "select ..."`. Note the `--` it specifies that no options follow for the command, so everything should be interpreted as normal arguments (even if they start with a `-`).

Reply:
@JelteF weirdly, this is what happens:

```sh
ssh -l ha-admin -t `vm_ip app` -- psql "\"$APP_DB_URI\"" -c 'SELECT count(*) FROM foo;'
```
```
bash: -c: line 0: syntax error near unexpected token `('
bash: -c: line 0: `psql "postgres://etcetc" -c SELECT count(*) FROM foo;'
```